### DISCUSSION TITLE
ERROR Fatal Error occurred - Function 'calculateInversions' - ubuntu 22.04

### DISCUSSION THREAD

**Discussion by asenookei** [nikosT/Gisola#49] [General]
Hi @nikosT 

I have tried to install Gisola on a PC with Ubuntu 22.04. I encountered issues when running tests, and errors appeared. Please help me. My plan is to install Seiscomp on the same PC to receive data from 60 sensor sites for inversion setup.

cd Gisola/src/core/inversion;
make MODEL=openmp 
2024-06-19 03:34:06,231 INFO Function 'calculateInversions' executed 2.48 seconds
2024-06-19 03:34:06,232 ERROR Fatal Error occurred
Traceback (most recent call last):
  File "/home/sysop/Gisola/src/./gisola.py", line 235, in <module>
    isola.gatherResults()
  File "/home/sysop/Gisola/src/config.py", line 181, in inner
    func(*args, **kwargs)
  File "/home/sysop/Gisola/src/isola.py", line 455, in gatherResults
    with open(os.path.join(workdir,'inversions', inversion,'inv1.dat')) as _:
FileNotFoundError: [Errno 2] No such file or directory: '../test/results/2020/noa2020vipzs/2024-06-19T03:32:20.066089Z/inversions/0.0.0.0.0.0/inv1.dat'
2024-06-19 03:34:06,233 INFO Moving to next event, if any

[log.txt](https://github.com/user-attachments/files/15895119/log.txt)

**Comment by nikosT** [comment]
Hi @asenookei, according to the log, it seems that you need to run the command `ulimit -s unlimited` just before the computations as defined here: https://github.com/nikosT/Gisola/wiki/Installation, in order to avoid the `Segmentation fault (core dumped)`. Could you check if that works for you?

You then need to source it in your `.bashrc` or `.bash_profile` to make it persistent.

**Reply by asenookei** [reply]
Thank you @nikosT for responding. The application is now running smoothly. I still have a few questions:

1. How can sensor usage for local earthquakes be optimized, where sensors operate at 40Hz frequency with channels SHZ, SHE, SHN, and several strong motion sensors? What is the optimal configuration for local earthquakes? I have attached the configuration that I am using.
2. Why aren't all results appearing in index.html? How can this issue be resolved?
note : version 1.2.1

[config_dev.txt](https://github.com/user-attachments/files/15951961/config_dev.txt)

![Screenshot from 2024-06-24 15-17-07](https://github.com/nikosT/Gisola/assets/11732743/1cc43845-70e5-4079-8098-334ff13ba0b9)

index.html
![Screenshot from 2024-06-24 15-16-35](https://github.com/nikosT/Gisola/assets/11732743/be7869f7-4132-45f0-8106-33faffa73858)
![Screenshot from 2024-06-24 15-15-50](https://github.com/nikosT/Gisola/assets/11732743/e7c0e5de-b909-40ae-9124-88efc6c5aaa2)

**Reply by nikosT** [reply]
Dear @asenookei,
That's great news! Concerning the first question. You should follow the suggested options used in NOA and they are defined here: https://github.com/nikosT/Gisola/wiki/Configuration#national-observatory-of-athens-noa-case-study, and particularly, check the file: https://bbnet.gein.noa.gr/gisola/configuration/config_ppsd_exp.yaml. Perhaps, prof @esokos could suggest more options on that, when using SH* channels. About the second question, there is a variable, specifically, in the config file here: https://github.com/nikosT/Gisola/blob/main/src/config.yaml#L427. You, can specify there the quality threshold of MT solution that you want to be able to appear. This is useful if you do not want all automatic calculations to appear before reviewing them.  If you want all of them to appear you can set it to D4. In any case if you attach at the main URL the suffix `?show=1"`, for instance `".../2024/index.html?show=1"`, all MT solutions will appear even if you have configured a strict quality threshold.

**Comment by esokos** [comment]
Dear @asenookei ,
i am not sure i understand your question. Is your network based on short period stations only ? what do you mean by "sensors operate at 40Hz frequency " is this the sampling frequency ? What is the frequency band that your sensors can record reliably ? What do you mean by local earthquakes ? what are the typical epicentral distances.? as regards the use of strong motion data .. what is the low freq noise?? for small events i am afraid it will be quite large ...

From the config file i see that you have specified some rules that use very low frequencies e.g.     - [3.5, 5.5, [0.04, 0.05, 0.08, 0.09]]     - [5.6, 5.9, [0.02, 0.03, 0.06, 0.07]]     - [6.0, 7.0, [0.01, 0.02, 0.05, 0.06]] ... this doesn't look as short period sensors !

In principle BEFORE trying to automate the procedure you should use the manual ISOLA on some events of your interest and this will give you the "bounds" of the variables you'll have to put in config. It will also give you info about the quality of your data.

**Reply by asenookei** [reply]
Dear @nikosT @esokos 

We are using archived data with sampling rates of 40Hz, 50Hz, and 100Hz, where channels use SH*, BH*, and HN*. I have modified the frequency configuration from [3.5, 5.5, [0.04, 0.05, 0.08, 0.09]] - [5.6, 5.9, [0.02, 0.03, 0.06, 0.07]] - [6.0, 7.0, [0.01, 0.02, 0.05, 0.06]] from the template configuration and changed it to:

Frequency:

[3.0, 3.5, [0.03, 0.04, 0.05, 0.1]]
[3.5, 4.5, [0.03, 0.04, 0.05, 0.1]]
[4.6, 5.5, [0.03, 0.04, 0.05, 0.1]]
[5.6, 7.0, [0.03, 0.04, 0.05, 0.1]]
[7.1, 9.0, [0.03, 0.04, 0.05, 0.1]]
Please check if this is correct for frequencies of 40Hz, 50Hz, and 100Hz.

For local earthquakes, it means earthquakes in the area around the sensor as shown in the picture above.

I have attached the results of the new configuration; please check if it is good.
[output.zip](https://github.com/user-attachments/files/15985610/output.zip)
[config_dev.txt](https://github.com/user-attachments/files/15985626/config_dev.txt)

**Comment by esokos** [comment]
Hello @asenookei,

this is not correct, the sampling frequency is not critical for the inversion frequency band. The freq band is set according to event magnitude (smaller magnitude --> higher frequencies) and is also controlled by the sensor type, the noise and the crustal model. You can read the ISOLA papers here https://github.com/esokos/isola/tree/master/Isola_Release_2024, start with ISOLA_bookchapter_english.pdf.

As i have written before, i would suggest to use the manual ISOLA (or some other MT inversion code) to understand the limitations of your data, crustal model etc and then transfer this knowledge to Gisola.

I saw briefly the files in the output.zip, what is the magnitude of this event ? i see only 2 stations that depict some good S/N ratio. I don't think you can use this data.

**Comment by asenookei** [comment]
Thank you Prof @esokos , I'll try your suggestion
